### PR TITLE
Fix scaling of phi in line with empirical bayes approach

### PR DESCRIPTION
**Pull request by Alanocallaghan**
- scaling of `mu.mu` in `.EmpiricalBayesMu` to match the scale given by spike-ins
- scaling of `mu0` in `.BASiCS_MCMC_Start` to match the scale given by spike-ins
when using an EB prior for mu
- lower minimum tolerance `mintol_mu` (1e-5 instead of 1e-3) as a default value
in `.BASiCS_MCMC_ExtraArgs`

### CODE REVIEW

**Comment by Alanocallaghan** [comment]
Seems to converge fairly well and give similar ESS across a bunch of datasets

File: R/utils_MCMC.R [hunk 1]
```diff
@@ -270,6 +270,17 @@
 
   # Correction for those genes with total count = 0
   aux <- ifelse(aux == 0, min(aux[aux > 0]), aux)
+  
+  # Extra scaling if the data has spike-ins
+  if(length(altExpNames(Data)) > 0) {
```
Comment (Alanocallaghan) [inline R/utils_MCMC.R:275]: this assumes that if altExps exist then WithSpikes is true which isn't necessarily the case

Comment (Alanocallaghan) [reply R/utils_MCMC.R:275]: Solution is to add a with_spikes arg here and then add one also to BASiCS_PriorParam. Or to do a more detailed check here in release, and in devel to handle arguments properly

Comment (Alanocallaghan) [reply R/utils_MCMC.R]: Actually can get around this by setting mu.mu to be NULL in the PriorParam constructor and then setting it later based on the value of PriorParam$PriorMu, WithSpikes etc

File: DESCRIPTION [hunk 2]
```diff
@@ -1,8 +1,8 @@
 Package: BASiCS
 Type: Package
 Title: Bayesian Analysis of Single-Cell Sequencing data
-Version: 2.1.16
-Date: 2020-10-22
+Version: 2.2.1
```
Comment (catavallejos) [inline DESCRIPTION:4]: Should we 2.3.1 to match version numbers in devel. We can use 2.2.1 when creating the fix for the release branch.

File: NEWS [hunk 3]
```diff
@@ -1557,3 +1557,10 @@ empirical Bayes prior is assigned to mu
 - Fixes small bug in `.BASiCS_MCMC_Start`
 - Updates unit tests linked to the change in `.BASiCS_MCMC_Start`
 - Adds fixed seeds to `test_divide_and_conquer.R`
+
+version 2.2.1 (2020-12-08)
```
Comment (catavallejos) [inline NEWS:1561]: version number change as suggested above

File: DESCRIPTION (hunk 2 above)
Comment (Alanocallaghan) [reply DESCRIPTION:4]: This is the fix for devel - I pulled from bioc

Comment (catavallejos) [reply DESCRIPTION:4]: Maybe that was from an old version? Current version in devel is 2.3.0: http://bioconductor.org/packages/devel/bioc/html/BASiCS.html

Comment (Alanocallaghan) [reply DESCRIPTION:4]: Sorry typo. This is meant to be the release patch - hence the target branch

Comment (catavallejos) [reply DESCRIPTION:4]: The `RELEASE_3_12` branch is 2.2.0

Comment (catavallejos) [reply DESCRIPTION:4]: ok, got it! Sorry, I didn't read the PR properly. I will merge now.

File: NEWS (hunk 3 above)
Comment (catavallejos) [reply NEWS:1561]: comment to be ignored, see discussion above

**Pull request merged**